elements/pfe-clipboard/src/pfe-clipboard.js

```diff
@@ -0,0 +1,173 @@
+    };
+  }
+
+  get templateUrl() {
+    return "pfe-clipboard.html";
+  }
+
+  get styleUrl() {
+    return "pfe-clipboard.scss";
+  }
+
+  static get events() {
+    return {
+      copied: `${this.tag}:copied`
+    };
+  }
+
+  // Declare the type of this component
+  static get PfeType() {
+    return PFElement.PfeTypes.Content;
+  }
+
+  static get properties() {
+    return {
+      iconHidden: {
+        title: "Icon Hidden",
+        type: Boolean,
+        attr: "icon-hidden"
+      }
+    };
+  }
+
+  static get slots() {
+    return {
+      icon: {
+        title: "Icon",
+        slotName: "pfe-clipboard--icon",
+        slotClass: "pfe-clipboard__icon",
+        slotId: "icon"
+      },
+      default: {
+        title: "Text",
+        slotName: "pfe-clipboard--text",
+        slotClass: "pfe-clipboard__text",
+        slotId: "text"
+      },
+      textSuccess: {
+        title: "Text Success",
+        slotName: "pfe-clipboard--text--success",
+        slotClass: "pfe-clipboard__text--success",
+        slotId: "text--success"
+      }
+    };
+  }
+
+  constructor() {
+    super(PfeClipboard, { type: PfeClipboard.PfeType });
+
+    this._icon = this.shadowRoot.querySelector(`#icon`);
+    this._text = this.shadowRoot.querySelector(`#text`);
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+    // If you need to initialize any attributes, do that here
+
+    this.icon = this.querySelector(`[slot="${this.tag}--icon"]`);
+    this.text = this.querySelector(`[slot="${this.tag}--text"]`);
+    this.textSuccess = this.querySelector(`[slot="${this.tag}--text--success"]`);
+
+    // Add a slotchange listener to the lightDOM trigger
+    // this.icon.addEventListener("slotchange", this._init);
+
+    // Add accessibility attributes to treat this element as a button
+    this._containerElement = this.shadowRoot.querySelector(".pfe-clipboard__container");
+    this._containerElement.setAttribute("role", "button");
+    this._containerElement.setAttribute("tabindex", "0");
+
+    // Since this element as the role of button we are going to listen
+    // for click and as well as 'enter' and 'space' commands to trigger
+    // the copy functionality
+    this._containerElement.addEventListener("click", this._clickHandler.bind(this));
+    this._containerElement.addEventListener("keydown", this._keydownHandler.bind(this));
+  }
+
+  disconnectedCallback() {
+    this._containerElement.removeEventListener("click", this._clickHandler.bind(this));
+    this._containerElement.removeEventListener("keydown", this._keydownHandler.bind(this));
+  }
+
+  _clickHandler(event) {
+    // Execute the copy to clipboard functionality
+    this.copyURLToClipboard()
+      .then(url => {
+        // Emit event that lets others know the user has "copied"
+        // the button. We are also going to include the url that was
+        // copied.
+        this.emitEvent(PfeClipboard.events.copied, {
+          detail: {
+            url
```

Comment:
very nice!

Reply:
thanks!  yeah it's cool.  just a little url that bubbles up 🛁